### PR TITLE
[feat] Add a legend for the "gt box" and "est box" in the visualization function

### PR DESCRIPTION
**Pull request by Charmve**
[feat] Make an option whether or not to render with the legend
| display_legend <br> ``True`` |display_legend <br> ``False`` |
|--|--|
|image|image|

### CODE REVIEW

**Comment by whyekit-motional** [comment]
@Charmve pls include a short description (reason for the change, an example rendering before and after your change, etc.) for your PR

File: python-sdk/nuscenes/eval/detection/render.py [hunk 1]
```diff
@@ -87,6 +87,9 @@ def visualize_sample(nusc: NuScenes,
         if box.score >= conf_th:
             box.render(ax, view=np.eye(4), colors=('b', 'b', 'b'), linewidth=1)
 
+    # Add legend
```
Comment (whyekit-motional) [inline python-sdk/nuscenes/eval/detection/render.py:90]: Add a full-stop at the end of the sentence (this is in line with the style for `nuscenes-devkit`)

File: python-sdk/nuscenes/eval/detection/render.py [hunk 2]
```diff
@@ -87,6 +87,9 @@ def visualize_sample(nusc: NuScenes,
...
+    ax.legend(['GT Box', 'EST Box'], loc='upper right')
```
Comment (whyekit-motional) [inline python-sdk/nuscenes/eval/detection/render.py:91]: `GT Box` --> `GT` and `EST Box` --> `EST`

Comment (whyekit-motional) [inline python-sdk/nuscenes/eval/detection/render.py:91]: Let's make this an option, so that users can choose whether or not to render with the legend

File: python-sdk/nuscenes/eval/detection/render.py [hunk 3]
```diff
@@ -28,6 +28,7 @@ def visualize_sample(nusc: NuScenes,
                      conf_th: float = 0.15,
                      eval_range: float = 50,
                      verbose: bool = True,
+                     display_legend: bool = True,
```
Comment (whyekit-motional) [inline python-sdk/nuscenes/eval/detection/render.py:31]: `display_legend` should be `False` by default to be backwards-compatible

File: python-sdk/nuscenes/eval/detection/render.py [hunk 4]
```diff
@@ -95,7 +101,7 @@ def visualize_sample(nusc: NuScenes,
     # Show / save plot.
     if verbose:
         print('Rendering sample token %s' % sample_token)
-    plt.title(sample_token)
+        plt.title(sample_token)
```
Comment (whyekit-motional) [inline python-sdk/nuscenes/eval/detection/render.py:104]: Why is this change needed?

**Comment by whyekit-motional** [comment]
@Charmve in your PR description, pls include two example renderings: one when the `display_legend` flag is enabled, and one when the `display_legend` flag is disabled

File: python-sdk/nuscenes/eval/detection/render.py (hunk 4 above)
Comment (Charmve) [reply python-sdk/nuscenes/eval/detection/render.py:104]: Users  uses verbose to determine whether to display the title

File: python-sdk/nuscenes/eval/detection/render.py (hunk 3 above)
Comment (Charmve) [reply python-sdk/nuscenes/eval/detection/render.py:31]: have done

**Comment by Charmve** [comment]
> @Charmve in your PR description, pls include two example renderings: one when the `display_legend` flag is enabled, and one when the `display_legend` flag is disabled

have done

File: python-sdk/nuscenes/eval/detection/render.py (hunk 4 above)
Comment (whyekit-motional) [reply python-sdk/nuscenes/eval/detection/render.py:105]: `verbose` should only be used to control `print`s - let's undo this change

Comment (Charmve) [reply python-sdk/nuscenes/eval/detection/render.py]: ok, and it has done

**Review by whyekit-motional — APPROVED** [review]
Thanks for feature @Charmve!
